Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -248,7 +248,6 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
             case .reconnected:
                 self.tunnelHealth.isHavingConnectivityIssues = false
                 self.notificationsPresenter.showReconnectedNotification()
-                self.reasserting = false
```

Comment:
This was causing the issues you were seeing.  The reassertion logic should really only be touched by the wireguard adapter.